### PR TITLE
Add New Field Network Attachment in Compute Instance Template Resource

### PR DESCRIPTION
**Pull request by daveavi**
I have made changes to the magic-modules repo to support a new field, network attachment resource for the compute instance template resource. It will enable creating compute workloads with PSC (Private Service Connect) Network Interfaces.

Fixes: https://github.com/hashicorp/terraform-provider-google/issues/14615

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
google_compute_instance_template: Added field `network_attachment` to `google_compute_instance_template`
```

### CODE REVIEW

File: mmv1/Gemfile.lock
Comment (trodge) [inline mmv1/Gemfile.lock:1]: Could you revert this file?

**Review by trodge — CHANGES_REQUESTED** [review]

File: mmv1/Gemfile.lock
Comment (daveavi) [reply mmv1/Gemfile.lock:1]: Done

File: mmv1/third_party/terraform/utils/compute_instance_helpers.go.erb [hunk 1]
```diff
@@ -348,11 +358,29 @@ func expandNetworkInterfaces(d tpgresource.TerraformResourceData, config *transp
...
+		if networkAttachment != "" {
+			if network != "" {
+				return nil, fmt.Errorf("Cannot have a network provided with networkAttachment given that networkAttachment is associated with a network already")
+			}
+			if subnetwork != "" {
+				return nil, fmt.Errorf("Cannot have a subnetwork provided with networkAttachment given that networkAttachment is associated with a subnetwork already")
+			}
+		}
+
+		<% else -%>
```
Comment (trodge) [inline mmv1/third_party/terraform/utils/compute_instance_helpers.go.erb:378]: This statement is missing an `end`.

**Review by trodge — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/utils/compute_instance_helpers.go.erb (hunk 1 above)
Comment (daveavi) [reply mmv1/third_party/terraform/utils/compute_instance_helpers.go.erb:378]: Done

File: mmv1/third_party/terraform/utils/bootstrap_test_utils.go [hunk 2]
```diff
@@ -9,9 +9,9 @@ import (
 	"testing"
 	"time"
 
-	"github.com/hashicorp/terraform-provider-google/google/acctest"
-	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
-	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+	"github.com/hashicorp/terraform-provider-google-beta/google-beta/acctest"
```
Comment (trodge) [inline mmv1/third_party/terraform/utils/bootstrap_test_utils.go:12]: This needs to import from the ga version.

**Review by trodge — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/utils/bootstrap_test_utils.go (hunk 2 above)
Comment (daveavi) [reply mmv1/third_party/terraform/utils/bootstrap_test_utils.go]: Done

**Pull request closed without merging**